### ISSUE TITLE
Canu snapshot from October 21 still yields much worse NG50 than v1.6

### ISSUE THREAD

**Issue by MattStata** [marbl/canu#704]
Hi again,

My assemblies that I started running with the (at the time) newest snapshot of Canu from Oct 21 finally finished, and in both cases the NG statistics are significantly worse than the one assembly that I managed to finish with the regular v1.6 release.

My first run with v1.6 resulted in NG50=1,704,491 after consensus for the one species that finished successfully, while both my (potentially buggy) second run and most recent run both resulted in NG50=767,773.

I could just use the successful first run for this species, but for the other species I am trying to assemble the regular v1.6 release  stalled endlessly and didn't finish, even when I deleted the last few output steps and restarted.

I am wondering if you guys have any idea why the Oct 21 snapshot yielded such a different NG50, and whether this issue might be corrected now so I could use the current snapshot and get a result more like what the original 1.6 release provides?  I could try running again with regular v1.6 but it did result in lots of stalling and required a lot of deleted output and restarting.  The newer snapshots seem to fix those issues but at the expense of poorer assembly stats...

Any suggestions for how I might best proceed would be greatly appreciated.

Cheers,
Matt

PS I am attaching all three report files.

[APRO.thirdrun.report.txt](https://github.com/marbl/canu/files/1472624/APRO.thirdrun.report.txt)
[APRO.firstrun.report.txt](https://github.com/marbl/canu/files/1472625/APRO.firstrun.report.txt)
[APRO.secondrun.report.txt](https://github.com/marbl/canu/files/1472626/APRO.secondrun.report.txt)

**Comment by skoren** [comment]
Yes, the tip still has some bugs/issues in read correction. You can see the human assembly result is worse with the 9/23 version than the 1.6 release. For now, I'd suggest sticking with the 1.6 release for assembly. You can correct with 1.6 and assemble the corrected reads with the latest tip as well.

**Comment by skoren** [comment]
The issue is some changes to the base calling algorithm which generates the corrected read. This should not have affected functionality but inadvertently did so until that is fixed, I would not recommend using the tip.

**Comment by brianwalenz** [comment]
Fixed.

The problem was a quirk in the way the evidence reads were mapped to the target read, which, I think, was mixing repeat copies together during read correction.  Then, when contigs were created, fewer repeats could be resolved (because we blurred, say, two slightly distinct repeat copies together) leading to lower contig sizes.

In our human tests, contig sizes are still a bit smaller, but it's only from a small handful of contig breaks.
